Task-8

olgoncharov opened the pull request:
1. Реализован метод `MultiCreateTip` для множественного добавления сущностей в базу батчами
2. Реализован метод `UpdateTip` для обновления сущности
3. Добавлены **Producer**-ы для создания **CUD** событий
4. Добавлены метрики на кол-во успешных вызовов для **CUD** операций

olgoncharov committed:
1. Implemented handlers for update tips and multicreate
2. Added kafka producer for CUD
3. Added metrics

dmzkrsk commented on api/ocp-tip-api/ocp-tip-api.proto:
```diff
@@ -49,6 +65,26 @@ message CreateTipV1Response {
 
+message MultiCreateTipV1Request {
+  repeated CreateTipV1Request Tips = 1;
+  uint64 BatchSize = 2;
```
Размером батча лучше пусть управляет приложение, а не клиент

olgoncharov replied:
Убрал этот параметр из запроса

dmzkrsk commented on api/ocp-tip-api/ocp-tip-api.proto:
```diff
@@ -49,6 +65,26 @@ message CreateTipV1Response {
+
+message MultiCreateTipV1Response {
+  repeated uint64 Ids = 1;
+  repeated CreateTipV1Request NotCreatedTips = 2;
```
Лучше сделать явно тип для `*Response`

olgoncharov replied:
Сделал

dmzkrsk commented on internal/metrics/metrics.go:
```diff
@@ -0,0 +1,21 @@
+
+var cudCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "tips_cud_operations_count",
```
Есть гайд, как лучше называть метрики https://prometheus.io/docs/practices/naming/

olgoncharov replied:
Поправил

olgoncharov committed:
Fixed issues after review:

1. Removed `BatchSize` parameter from multi create request body
2. Created new type for multi create response
3. Renamed metric
